changelogs/fragments/273-add_multiple_options_with_same_name_to_ini_file.yml

```diff
@@ -0,0 +1,6 @@
+minor_changes:
+  - add the ability to 'add' or 'remove' single option=value entries
+  - add the ability to define multiple options with the same name but different values
+
+trivial:
+  - restructure tests for later extension
```

Comment:
Please note that the bugfix should better go into a separate PR, so it can be backported to stable-2 as well (if it doesn't only affect stable-3 to which new features are backported as well).

Reply:
I will split those two changes into separate PR.

Reply:
Also please note that #273 and #1204 are not really bugs, but feature requests. So "fixing" them is not a bugfix, but a minor change.

Reply:
One of the two is labeled as 'bug'...

Generally speaking: I need the feature in production, so I do not really care if they go as bug or as feature. I just need to know if I should change the change log entry to bugfix or keep it as minor.

Reply:
looking at #273 and #1204, for me the first reports a bug (inconsistent results, depending on the initial state of the file - the order of values for a same option), even if the explicit wish is to "not clobber existing options"; and the second is clearly a feature request. Anyway I think the bug could be fixed more easily by adding the feature :)

@ziegenberg , I suggest you keep all as minor changes, as long as it adds a feature without modifying the existing interface (options names and types, results names and types, default behaviour, mandatory options). @felixfontein , do you agree ?

Whatever is decided about `bugfixes` or `minor_changes` status, changelog entries should be formated like:
```yaml
  - module_name - description of the change (issue(s) or pull request url(s)).
```

Reply:
> @ziegenberg , I suggest you keep all as minor changes, as long as it adds a feature without modifying the existing interface (options names and types, results names and types, default behaviour, mandatory options).

That's why I first edited the tests (first commit all Tests are good with no code changes) and in the commits afterwards I changed the code and the tests are still good (plus I added a bunch of new test). That way it should be guaranteed, that my changes / the new feature / bugfix did not alter existing behaviour.

Reply:
> looking at #273 and #1204, for me the first reports a bug (inconsistent results, depending on the initial state of the file - the order of values for a same option), even if the explicit wish is to "not clobber existing options";

I would disagree that this is a bug. The module does not handle INI files with repeated keys, so the current behavior - check first occurence, replace all occurences if the first one does not match - is consistent. It's just not what everyone expects.

> @ziegenberg , I suggest you keep all as minor changes, as long as it adds a feature without modifying the existing interface (options names and types, results names and types, default behaviour, mandatory options). @felixfontein , do you agree ?

Sounds good to me.